python/lsst/ap/association/loadDiaCatalogs.py

```diff
@@ -192,6 +197,35 @@ def loadDiaSources(self, diaObjects, pixelRanges, dateTime, apdb):
                              inplace=True)
         return diaSources.replace(to_replace=[None], value=np.nan)
 
+    @pipeBase.timeMethod
+    def loadDiaForcedSources(self, diaObjects, dateTime, apdb):
+        """Load DiaObjects from the Apdb based on their HTM location.
+
+        Parameters
+        ----------
+        pixelRanges : `tuple` [`int`]
+            Ranges of pixel values that cover region of interest.
+        apdb : `lsst.dax.apdb.Apdb`
+            Database connection object to load from.
+
+        Returns
+        -------
+        diaObjects : `pandas.DataFrame`
+            DiaObjects loaded from the Apdb that are within the area defined
+            by ``pixelRanges``.
+        """
+        if len(diaObjects) == 0:
+            # If no diaObjects are available return an empty DataFrame with
+            # the the column used for indexing later in AssociationTask.
+            diaForcedSources = pd.DataFrame(columns=["diaObjectId"])
+        else:
+            diaForcedSources = apdb.getDiaForcedSources(
+                diaObjects.loc[:, "diaObjectId"],
```

Comment:
Should the object IDs be trimmed down to just unique values? Maybe it's already guaranteed to be unique, but what happens if there are object IDs that appear multiple times, are forced sources duplicated in the result here?

Reply:
So, the daiObjectIds are duplicated for sure but the row values are unique so I don't think there will be any duplicated data here. I believe pandas groups together the duplicated values, but what I can do is specifically add the diaForcedSourceId here to make sure this is what is happening and get an ordering for the diaForcedSources in the alert packet.